dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/ProgramBuilder.java

```diff
@@ -65,19 +71,27 @@ public Program build(){
         return program;
     }
 
-    public void initThread(String name, int id){
-        if(!threads.containsKey(id)){
+    public void initThread(String name, int id) {
+        if(!functions.containsKey(id)){
             Skip threadEntry = EventFactory.newSkip();
```

Comment:
> entry sounds like a better name since this is not necessary a thread.

It is necessarily a thread that gets created. There is a separate `initFunction` method that creates functions.

> Not sure if it make sense in the current state of this PR, but shouldn't we at some point add FunctionEntry and FunctionExit nodes (i.e. events)?

Why? I don't think functions need any of that, because (1) you can have a empty (= no body) functions and (2) we have a `Return` event to exit functions.

Reply:
In a CFG you would normally have entry/exit nodes. The `Return` event is exactly what I had in mind (I wrote the comment before going over all changes). Not sure if having a specialized `Entry extends Skip` would make sense or bring any advantage.

Reply:
Currently, all function bodies start with a `$bb0` label. And I think this will also holds true for LLVM code.
At least for now, a dedicated entry node just has disadvantages if you allow for empty-bodied functions. Because on an empty function, `f.appendEvent(e)` would need to implicitly create an entry event before appending `e`.

Reply:
Actually, now that I think about it, we will probably need a marker of sorts to distinguish between "empty body" and "no body". Maybe a dedicated `FunctionEntry` event could serve this purpose. But I would do this in a PR that does add proper support for functions (i.e. part 2 or 3, depending on how difficult it gets :)).